framework/tests/field-type-inference/DeclaredTypeTest.java

```diff
@@ -0,0 +1,49 @@
+import tests.fieldtypeinference.qual.*;
+
+public class DeclaredTypeTest {
+    // Case where the declared type is a supertype of the refined type.
+    private @Top int privateDeclaredField;
+    public @Top int publicDeclaredField;
+
+    // The type of privateDeclaredField is refined to @FieldTypeInferenceBottom
+    // because of the first method call in the method below.
+    void assignFieldsToBottom() {
+        privateDeclaredField = getBottom();
+        publicDeclaredField = getBottom();
+    }
+
+    // Testing the refinement above.
+    void testFields() {
+        expectsBottom(privateDeclaredField);
+        // public fields are not refined.
+        //:: error: (argument.type.incompatible)
+        expectsBottom(publicDeclaredField);
+    }
+
+    // Case where the declared type is a subtype of the refined type.
+    private @FieldTypeInferenceBottom int privateDeclaredField2;
+    public @FieldTypeInferenceBottom int publicDeclaredField2;
+
+    // The refinement cannot happen and an assignemnt type incompatible error occurs.
+    void assignFieldsToTop() {
+        //:: error: (assignment.type.incompatible)
+        privateDeclaredField2 = getTop();
+        //:: error: (assignment.type.incompatible)
+        publicDeclaredField2 = getTop();
```

Comment:
Should you also have an assignment with RHS getBottom() just to make sure that doesn't issue any errors?